### PR TITLE
Chapter 12: Tools in Visual Studio

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: content/vstools/reading/nuget/index.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+1. Select “Manage NuGet Packages” from the dropdown menu
+1. The resulting window will show you a catalog of software packages you may add to your project
+
+### Managing NuGet Packages
+
+The following documentation can help you become more familiar with downloading and managing NuGet packages.
+
+**Windows Users:** [Follow this guide](https://learn.microsoft.com/en-us/nuget/consume-packages/install-use-packages-visual-studio).
+
+**Mac Users:** [Use this guide](https://learn.microsoft.com/en-us/visualstudio/mac/nuget-walkthrough?toc=%2Fnuget%2Ftoc.json&view=vsmac-2022).
```
Comment (codinglikeagirl42) [inline content/vstools/reading/nuget/index.md:39]:
```suggestion
[**Windows Users: Follow this guide.**](https://learn.microsoft.com/en-us/nuget/consume-packages/install-use-packages-visual-studio).

[**Mac Users: Follow this guide.**](https://learn.microsoft.com/en-us/visualstudio/mac/nuget-walkthrough?toc=%2Fnuget%2Ftoc.json&view=vsmac-2022).
```

File: content/vstools/reading/debugger/index.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+     currently sitting on, as well as the line above.
+   1. [Locals](https://learn.microsoft.com/en-us/visualstudio/debugger/autos-and-locals-windows?view=vs-2022) pane shows the value of local variables and parameters within the program being debugged.
+   1. Add variables, parameters, and expressions to the [Watch](https://learn.microsoft.com/en-us/visualstudio/debugger/debugger-feature-tour?view=vs-2022#set-a-watch) pane to monitor their values while debugging.
+   1. [Call Stack](https://learn.microsoft.com/en-us/visualstudio/debugger/how-to-use-the-call-stack-window?view=vs-2022) pane displays the record of the methods that have been called in the program being debugged.
+   - View a list of breakpoints in the *Breakpoints* pane. You may also disable and enable breakpoints from here.
+1. Debugger [Code Stepping](https://learn.microsoft.com/en-us/visualstudio/debugger/navigating-through-code-with-the-debugger?view=vs-2022&tabs=csharp#code-stepping) Buttons:
+   1. [Step over](https://learn.microsoft.com/en-us/visualstudio/debugger/debugger-feature-tour?view=vs-2022#step-over-code-to-skip-functions) button moves debugger to the next line to be executed within a method.
```
Comment (codinglikeagirl42) [inline content/vstools/reading/debugger/index.md:29]: All these links seem to be for VS, do they also work for VS for Mac?

image

Comment (codinglikeagirl42) [reply content/vstools/reading/debugger/index.md:29]: I see there is another section on line 37 -> Extra tips for Mac users.  Maybe make a note around line 16, that this has been include.

File: content/vstools/reading/debugger/index.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+## Visual Studio for Mac Extra Tips
+
+1. Your IDE may not default to *Debug* mode. To select for it, in the top menu, select *Debug > Start Debugging*.
+1. To view the debugging panes, select *Debug* in the top menu and scroll down to *Windows*. Select 
+   the items you wish to view, ie. *Breakpoints*, *Watch*, etc.
+1. To add conditions to a breakpoint, right click on the breakpoint and select *Edit Breakpoint*. From 
+   menu that opens, use the *Advanced Conditions* section to set your conditions for when you want the 
+   breakpoint to be executed.
+1. For more information on using the Debugger in Visual Studio for Mac, check out this [guide](https://learn.microsoft.com/en-us/visualstudio/debugger/using-breakpoints?view=vs-2022).
```
Comment (codinglikeagirl42) [inline content/vstools/reading/debugger/index.md:45]: This guide goes to a VS article, not a VS for Mac article. Does this particular article work better for Mac users?

**Review by codinglikeagirl42 — CHANGES_REQUESTED** [review]
Please update the Metadata on all pages.  Look at other suggestions and make any changes need.  Let me know if you have any questions.

File: content/vstools/reading/debugger/index.md (hunk 2 above)
Comment (gildedgardenia) [reply content/vstools/reading/debugger/index.md:29]: Done!

File: content/vstools/reading/debugger/index.md (hunk 3 above)
Comment (gildedgardenia) [reply content/vstools/reading/debugger/index.md]: Done!

**Review by codinglikeagirl42 — APPROVED** [review]
Looks good, Merge away!

**Pull request merged**